File: AspNetMvcUnleashed/0672329980_source/Chapter12/VB/MvcApplication1/MvcApplication1.Tests/Controllers/JillControllerTests.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports System.Web.Mvc

<TestClass()> _
Public Class JillControllerTests

    <TestMethod()> _
    Public Sub JillCanAccessIndex()
        ' Arrange
        Dim controller = New JillController()
        Dim principal = New FakePrincipal("Jill")

        ' Act
        Dim result = controller.Index(principal)

        ' Assert
        Assert.IsInstanceOfType(result, GetType(ViewResult))
    End Sub

    <TestMethod()> _
    Public Sub JackCannotAccessIndex()
        ' Arrange
        Dim controller = New JillController()
        Dim principal = New FakePrincipal("Jack")

        ' Act
        Dim result = controller.Index(principal)

        ' Assert
        Assert.IsInstanceOfType(result, GetType(HttpUnauthorizedResult))
    End Sub


End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter12/CS/MvcApplication1/MvcApplication1.Tests/Controllers/JillControllerTests.cs
using System.Web.Mvc;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using MvcApplication1.Controllers;

namespace MvcApplication1.Tests.Controllers
{
    [TestClass]
    public class JillControllerTests
    {

        [TestMethod]
        public void JillCanAccessIndex()
        {
            // Arrange
            var controller = new JillController();
            var principal = new FakePrincipal("Jill");

            // Act
            var result = controller.Index(principal);

            // Assert
            Assert.IsInstanceOfType(result, typeof(ViewResult));

        }

        [TestMethod]
        public void JackCannotAccessIndex()
        {
            // Arrange
            var controller = new JillController();
            var principal = new FakePrincipal("Jack");

            // Act
            var result = controller.Index(principal);

            // Assert
            Assert.IsInstanceOfType(result, typeof(HttpUnauthorizedResult));

        }

    
    }




}
